src/NHibernate/SqlCommand/SqlSelectBuilder.cs

```diff
@@ -291,6 +298,19 @@ public SqlString ToSqlString()
 			return sqlBuilder.ToSqlString();
 		}
 
+		private string GetForUpdateString()
+		{
+			if (!Dialect.SupportsOuterJoinForUpdate && hasOuterJoin)
+			{
+				if (Equals(lockMode, LockMode.Upgrade))
+					return Dialect.GetForUpdateString(aliasToLock);
```

Comment:
We still have a trouble here: databases not supporting `for update` on outer join and not supporting `for update` aliases. For such database the code here will keep a `for update` clause, and it will still fail.

So either a new property needs to be added for knowing if the dialect supports lock aliases (currently, if it does not, it yields the `for update` without aliases), and used here, or the for update should just yield here an empty string, removing the need for transmitting the alias by the way.

Reply:
Is there any DBMS that matches? I'd like to leave this logic for a later date when we need it.

Reply:
Maybe not, but see my update on #1565, it is definitely incorrect to systematically supply a table alias here. Some DB do take indeed a column list here, not a table alias. The correct code handling both cases is in `ForUpdateFragment`.